### PR TITLE
Update to work with Snapcraft 3.0+

### PR DESCRIPTION
**Pull request by xordspar0**
Before these changes, tmnationsforever would not build with recent
versions of snapcraft.

* Move contents of snap/ directory to root and delete the snap/ directory.
* Use the snapcraft core as a base.
* Remove no-system-libraries build attribute, which no longer exists.
* Update Wine download URLs.

### CODE REVIEW

**Comment by xordspar0** [comment]
Note that this gets the build working, but I'm not done getting the game to run yet. Unlike the current stable release, the game does not segfault on startup with this build. Instead, it gets through installation and then seems to get stuck at the "blue screen" virtual desktop. I'll keep working on that in a separate PR.

**Comment by xordspar0** [comment]
@flexiondotorg I know this project isn't a high priority, but do you have time to take a quick look?

File: snapcraft.yaml [hunk 1]
```diff
@@ -122,8 +119,8 @@ parts:
       WINE_VER="3.10.0"
 
       # wget and dpkg extract the wine debs
-      DEB_URLS="https://dl.winehq.org/wine-builds/ubuntu/pool/main/${WINE_REL}-${WINE_ARCH}_${WINE_VER}~xenial_${WINE_ARCH}.deb \
-                https://dl.winehq.org/wine-builds/ubuntu/pool/main/${WINE_REL}_${WINE_VER}~xenial_${WINE_ARCH}.deb"
+      DEB_URLS="https://dl.winehq.org/wine-builds/ubuntu/dists/xenial/main/binary-i386/${WINE_REL}-${WINE_ARCH}_${WINE_VER}~xenial_${WINE_ARCH}.deb \
```
Comment (diddledan) [inline snapcraft.yaml]:
```suggestion
      DEB_URLS="https://dl.winehq.org/wine-builds/ubuntu/dists/xenial/main/binary-${WINE_ARCH}/${WINE_REL}-${WINE_ARCH}_${WINE_VER}~xenial_${WINE_ARCH}.deb \
```

File: snapcraft.yaml [hunk 2]
```diff
@@ -122,8 +119,8 @@ parts:
...
+                https://dl.winehq.org/wine-builds/ubuntu/dists/xenial/main/binary-i386/${WINE_REL}_${WINE_VER}~xenial_${WINE_ARCH}.deb"
```
Comment (diddledan) [inline snapcraft.yaml]:
```suggestion
                https://dl.winehq.org/wine-builds/ubuntu/dists/xenial/main/binary-${WINE_ARCH}/${WINE_REL}_${WINE_VER}~xenial_${WINE_ARCH}.deb"
```

**Comment by xordspar0** [comment]
Thanks for the review! I made your suggested changes.

**Comment by popey** [comment]
Sorry for the delay. Looks like https://dl.winehq.org/wine-builds/ubuntu/pool/main/wine-devel-i386_3.10.0~xenial_i386.deb is no longer available. Is there a preferred alternative build we can use?

**Comment by xordspar0** [comment]
Yes, this PR includes an update to the URL to point to the right spot on the downloads server. One thing we could consider is adding https://dl.winehq.org/wine-builds/ubuntu as an additional repository and using APT instead of wget, but this works as it is right now.

**Comment by popey** [comment]
You're right, my bad, sorry. This is great, thanks for your fixes. I've tested on my Intel based laptop and it works great.

**Pull request merged**